### PR TITLE
Tidy/fix operation name in errors.

### PR DESCRIPTION
**Pull request by rhattersley**

### CODE REVIEW

File: lib/iris/analysis/maths.py [hunk 1]
```diff
@@ -545,14 +551,14 @@ def apply_ufunc(ufunc, cube, other_cube=None, new_unit=None, new_name=None,
...
-def _binary_op_common(operation_function, operation_noun, cube, other,
+def _binary_op_common(operation_function, operation_name, cube, other,
                       new_unit, dim=None, in_place=False):
     """
     Function which shares common code between binary operations.
 
     operation_function   - function which does the operation
                            (e.g. numpy.divide)
-    operation_noun       - the noun of the operation (e.g. 'division')
+    operation_name       - the noun of the operation (e.g. 'division')
```
Comment (QuLogic) [inline lib/iris/analysis/maths.py]: `noun`->`name`, `division`->`divide`, I think.

File: lib/iris/analysis/maths.py [hunk 2]
```diff
@@ -610,14 +616,14 @@ def _broadcast_cube_coord_data(cube, other, operation_noun, dim=None):
...
-                                 % (operation_noun, operation_noun))
+                                 % (operation_name, operation_name))
 
     if other.ndim != 1:
         raise iris.exceptions.CoordinateMultiDimError(other)
 
     if other.has_bounds():
         warnings.warn('%s by a bounded coordinate not well defined, ignoring '
-                      'bounds.' % operation_noun)
+                      'bounds.' % operation_name)
```
Comment (QuLogic) [inline lib/iris/analysis/maths.py]: `operation_name.capitalize()` maybe?

Comment (rhattersley) [inline lib/iris/analysis/maths.py]: I agree that it doesn't read very well as it stands - I'll have a go at improving it.

File: lib/iris/analysis/maths.py (hunk 1 above)
Comment (rhattersley) [inline lib/iris/analysis/maths.py]: I'll make this consistent with `_add_subtract_common()`.

**Comment by rhattersley** [comment]
Updated and passing tests again.

**Comment by rhattersley** [comment]
@dkillick - have you got the bandwidth to merge this?

**Comment by dkillick** [comment]
@rhattersley :warning: task switching! :warning: I'll take a look.

**Comment by rhattersley** [comment]
> @rhattersley :warning: task switching! :warning: I'll take a look.

It's not urgent! There's no need to drop what you're doing. Sorry for the confusion!

File: lib/iris/analysis/maths.py [hunk 3]
```diff
@@ -220,26 +220,32 @@ def subtract(cube, other, dim=None, ignore=True, in_place=False):
...
-    operation_noun       - the noun of the operation (e.g. 'subtraction')
-    operation_past_tense - the past tense of the operation (e.g. 'subtracted')
+    operation_name       - the public name of the operation (e.g. 'divide')
+    cube                 - the cube whose data is used as the first argument
+                           to `operation_function`
+    other                - the cube, coord, ndarray or number whose data is
+                           used as the second argument
+    dim                  - dimension along which to apply `other` if it's a
+                           coordinate that is not found in `cube`
+    ignore               - ignored! ;-)
```
Comment (dkillick) [inline lib/iris/analysis/maths.py]: Yes, very nice, but what is being ignored? Is it the operation, the operation name, cube, other, or does it mean that the operation is just not executed? :wink:

File: lib/iris/analysis/maths.py [hunk 4]
```diff
@@ -613,14 +619,14 @@ def _broadcast_cube_coord_data(cube, other, operation_noun, dim=None):
...
+                                 % (operation_name, operation_name))
 
     if other.ndim != 1:
         raise iris.exceptions.CoordinateMultiDimError(other)
 
     if other.has_bounds():
-        warnings.warn('%s by a bounded coordinate not well defined, ignoring '
-                      'bounds.' % operation_noun)
+        warnings.warn('Using {!r} with a bounded coordinate is not well '
+                      'defined: ignoring bounds.'.format(operation_name))
```
Comment (dkillick) [inline lib/iris/analysis/maths.py]: Pedantry alert: I think we should use a semi-colon, rather than a comma or a colon here.

```
, + : = ;
```
???

Comment (rhattersley) [inline lib/iris/analysis/maths.py]: :+1: for pedantry :wink:

For completeness have you considered using a `-`?

`Using 'divide' with a bounded coordinate is not well defined: ignoring bounds.`

`Using 'divide' with a bounded coordinate is not well defined; ignoring bounds.`

`Using 'divide' with a bounded coordinate is not well defined - ignoring bounds.`

File: lib/iris/analysis/maths.py [hunk 5]
```diff
@@ -149,7 +149,7 @@ def _assert_matching_units(cube, other, operation_noun):
     if cube.units != getattr(other, 'units', cube.units):
         raise iris.exceptions.NotYetImplementedError(
             'Differing units (%s & %s) %s not implemented' %
-            (cube.units, other.units, operation_noun))
+            (cube.units, other.units, operation_name))
```
Comment (dkillick) [inline lib/iris/analysis/maths.py]: I'm not sure this error message has been helped by this change:
```python
'Differing units (m & kg) add not implemented'
```

Maybe `Cannot %s differing units (%s & %s).format(operation_name, cube.units, other.units)`:

```python
'Cannot add differing units (m & kg)'
```

or something.

Comment (rhattersley) [inline lib/iris/analysis/maths.py]: > I'm not sure this error message has been helped by this change:

I've not made it worse either :stuck_out_tongue_winking_eye: I'm happy to use this opportunity to improve things though.

File: lib/iris/analysis/maths.py (hunk 4 above)
Comment (dkillick) [inline lib/iris/analysis/maths.py]: Indeed. Often semi-colons are replaced with dashes, but I think that would flow better if the clauses were reversed:

`Ignoring bounds - using 'divide' with a bounded coordinate is not well defined.`

But that's a bit much. 

For reference, it should be an endash (&ndash;) in such a case.

File: lib/iris/analysis/maths.py (hunk 3 above)
Comment (rhattersley) [inline lib/iris/analysis/maths.py]: Wow... just discovered that this keyword was already deprecated way back in [version 0.8](https://github.com/rhattersley/iris/blob/61ac271d425eb6037a8e9d3be2458f02f7de111d/lib/iris/analysis/maths.py#L284)!

File: lib/iris/analysis/maths.py (hunk 4 above)
Comment (rhattersley) [inline lib/iris/analysis/maths.py]: > For reference, it should be an endash (–) in such a case.

You've raised the bar again ... world class pedantry! :stuck_out_tongue_winking_eye: 

You win; I'll use a semi-colon.

**Pull request merged**